src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -771,6 +774,9 @@ public void deleteResultsAndSubmissionsOfParticipation(Long participationId, boo
         submissions.forEach(submission -> {
             if (submission instanceof ProgrammingSubmission) {
                 coverageReportRepository.deleteBySubmissionId(submission.getId());
+                // This clears the build log entries and deletes them from the corresponding table
+                ((ProgrammingSubmission) submission).setBuildLogEntries(List.of());
+                submissionRepository.save(submission);
                 buildLogStatisticsEntryRepository.deleteByProgrammingSubmissionId(submission.getId());
             }
```

Comment:
Maybe use Java 17 instance of:

if (submission instanceof ProgrammingSubmission programmingSubmission) {
				programmingSubmission.setBuildLogEntries(List.of());
                submissionRepository.save(programmingSubmission);
                buildLogStatisticsEntryRepository.deleteByProgrammingSubmissionId(programmingSubmission.getId());